### ISSUE TITLE
False positives for `find*` queries from `react-test-renderer`

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#673]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v5.7.2

### ESLint version

v8.25.0

### Node.js version

16.15.0

### package manager and version

n/a

### Operating system

n/a

### Bug description

[The `react-test-renderer` library provides several utils](https://reactjs.org/docs/test-renderer.html#testinstance) which match the `find*` query pattern: `findByType`, `findByProps`, `findAll`, `findAllByType`,  `findAllByProps`.

These methods are provided from a `react-test-renderer` instance, or any [React Native Testing Library query](https://callstack.github.io/react-native-testing-library/docs/api-queries#queries). They should be ignored, so they are not treated as Testing Library queries (since they actually aren't).

This should apply to all rules, so it must be implemented in the [`isQuery` internal helper](https://github.com/testing-library/eslint-plugin-testing-library/blob/9eaca4a427fe80ae7e0901069a1c0afb8a89afdf/lib/create-testing-library-rule/detect-testing-library-utils.ts#L356). This function should ignore the queries listed before if they were imported from `react-test-renderer` or `@testing-library/react-native`. Checking the imports from the latter it's important since `react-native-testing-library` queries return nodes with the mentioned utils, so it's possible to do something like `getByRole('modal').findByProps('bar')`, hence ignoring `react-test-renderer` utils if coming from `@testing-library/react-native`

### Steps to reproduce

This example extracted from [React docs](https://reactjs.org/docs/test-renderer.html) to cover the `react-test-renderer` import:

```js
import TestRenderer from 'react-test-renderer';

function MyComponent() {
  return (
    <div>
      <SubComponent foo="bar" />
      <p className="my">Hello</p>
    </div>
  )
}

function SubComponent() {
  return (
    <p className="sub">Sub</p>
  );
}

const testRenderer = TestRenderer.create(<MyComponent />);
const testInstance = testRenderer.root;

expect(testInstance.findByType(SubComponent).props.foo).toBe('bar');
expect(testInstance.findByProps({className: "sub"}).children).toEqual(['Sub']);
```

Example to cover the `@testing-library/react-native` import:

```js
import { render } from '@testing-library/react-native';

test('some test case', () => {
  render(<SomeComponent />);
  const falsePositive = screen.getByText('foo').findByProps({ testID: 'bar' })
})
```

It would be interesting to check the test cases added in [this PR](https://github.com/testing-library/eslint-plugin-testing-library/pull/703/files#diff-2889ef91637a07f60b716ad39b8e9b3c61da2b4433e62b899ea27797116a4463).

### Error output/screenshots

`findByType`, `findByProps`, and other `react-test-renderer` utils are reported by the plugin (e.g., `await-async-queries` complains about they must be awaited).

### ESLint configuration

n/a

### Rule(s) affected

All rules reporting on queries are affected, but `await-async-queries` is the most obvious one.

### Anything else?

This was originated by #671 and #703

### Do you want to submit a pull request to fix this bug?

No

**Comment by Mrblackey** [comment]
Couldn't we just get away with explicitly excluding `/^find(All)?By(Type|Props)$/`?

**Comment by Belco90** [comment]
> Couldn't we just get away with explicitly excluding `/^find(All)?By(Type|Props)$/`?

I'm afraid not because that would silence legit custom queries named like that.

**Comment by sjarva** [comment]
Just making a note to whomever will fix this bug, there's another issue related to `react-test-renderer`: #494
and it will be handy to re-use the new helper function that detects import from `react-test-renderer` in fixes to both issues 📝

**Comment by Mrblackey** [comment]
> Just making a note to whomever will fix this bug, there's another issue related to `react-test-renderer`: #494 and it will be handy to re-use the new helper function that detects imports from `react-test-renderer` in fixes to both issues 📝

Indeed. Does this function already exist? Is it capable of tracking the origin of the `act` call if it is re-exported from another file, instead of directly from `react-test-renderer`?

**Comment by sjarva** [comment]
> Indeed. Does this function already exist?

No, not yet.

**Comment by Belco90** [comment]
After @notjosh's clarification on #703 I'm repurposing this ticket, so we just exclude `find*` names coming from `react-test-renderer` as @Mrblackey originally suggested.

**Comment by Belco90** [comment]
Taking care of this one!

**Comment by Belco90** [comment]
I've closed my original PR #709 since such workaround could lead to other problems. I don't have enough time to implement the ideal fix, but I've updated the issue again with more details about how it should be done.

As a temporary workaround, `@testing-library/react-native` users can avoid the false positives by s[witching off the `custom-queries` reporting entirely](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migration-guides/v4.md#testing-librarycustom-queries).

I'll create a PR to mention this in the [Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting).

**Comment by abejfehr** [comment]
Sorry to be dense, but I'm seeing false positives for `testing-library/await-async-query` in React Native Testing Library with queries like `screen.container.findByType(MyComponent)`, which seemingly _does not_ return a Promise.

I've really tried reading the Troubleshooting section mentioned in [this comment](https://github.com/testing-library/eslint-plugin-testing-library/issues/673#issuecomment-1365269620), but I can't figure it out.

What would I have to add in my eslint configuration to exempt `findByType` and `findAllByType` from this rule?
